src/profile-logic/symbol-store.js

```diff
@@ -120,17 +124,24 @@ export function readSymbolsFromSymbolTable(
 // sum(chunk.map(computeValue)) <= maxValue or chunk.length == 1.
 // The array is allowed to contain elements which are larger than the maximum
 // value on their own; such elements will get a single chunk for themselves.
+// Additionally, each chunk is limited to contain no more than maxChunkLength
+// elements.
 function _partitionIntoChunksOfMaxValue<T>(
   array: T[],
   maxValue: number,
+  maxChunkLength: number,
   computeValue: T => number
 ): T[][] {
   const chunks = [];
   for (const element of array) {
     const elementValue = computeValue(element);
     // Find an existing chunk that still has enough "value space" left to
     // accomodate this element.
-    let chunk = chunks.find(({ value }) => value + elementValue <= maxValue);
+    let chunk = chunks.find(
+      ({ value, elements }) =>
+        value + elementValue <= maxValue &&
+        elements.length + 1 <= maxChunkLength
```

Comment:
optional nit: `elements.length < maxChunkLength` is both simpler and (I think, but this is subjective) more understandable in this case.

Reply:
When I was looking at this change, I actually found it easier to understand with `+ 1`, because it felt like it's more explicit that this is about the size "when we add another item to the chunk".

Reply:
Nazim is exactly right, that was how I was thinking about it. We have two comparisons here: "What happens if we add `elementValue` to `value`, are we still good? And what happens if we add 1 to `elements.length`, are we still good?"

Reply:
As I said this is optional :-) up to you